transport/http_util.go

```diff
@@ -159,6 +161,8 @@ func newHPACKDecoder() *hpackDecoder {
 			}
 		case ":path":
 			d.state.method = f.Value
+		case "content-type":
```

Comment:
I actually think content-type is intended for this purpose. Also there is no good place to specify this at higher level (you are against the proposal of putting it in MethodDesc and something similar) on server side.